src/jobs/deploy-ecs.yml

```
@@ -1,9 +1,133 @@
-description: >
+description: |
   This job deploys an application to aws elastic container service.
 
-executor: node/default
+parameters:
+  # Required Parameters
+  image-name:
+    type: string
+    description: |
+      Name of the docker image in the docker.io repository
+  tag:
+    type: string
+    description: |
+      The semantic version of the project trying to be deployed
+      https://semver.org/spec/v2.0.0.html
+  cluster-name:
+    type: string
+    description: |
+      The short name or full ARN of the cluster that hosts the service.
+  # Not Required Parameters
+  container-name:
+    type: string
+    description: |
+      Name of the container name defined in the ecs task-definition
+      (defaults to image-name)
+  service-name:
+    type: string
+    description: |
+      The name of the service to update.
+      (defaults to image-name)
+  family:
+    type: string
+    description: |
+      Name of the task definition's family.
+      (defaults to image-name)
+  isService:
+    type: boolean
+    default: false
+    description: |
+      True if the task-definition is deployed as a service on ECS. (Default is false)
+  aws-access-key-id:
+    type: env_var_name
+    default: MACHINE_AWS_ECS_AccessKey
+    description: |
+      AWS access key id for IAM role. Set this to the name of
+      the environment variable you will use to hold this
+      value, i.e. MACHINE_AWS_ECS_AccessKey.
+  aws-secret-access-key:
+    type: env_var_name
+    default: MACHINE_AWS_ECS_SecretKey
+    description: |
+      AWS secret key for IAM role. Set this to the name of
+      the environment variable you will use to hold this
+      value, i.e. $MACHINE_AWS_ECS_SecretKey.
+  aws-region:
+    type: env_var_name
+    default: AWS_REGION_N_VA
+    description: |
+      Env var of AWS region to operate in
+      (defaults to AWS_REGION_N_VA)
+  slack-mentions:
+    type: env_var_name
+    default: MENTIONS
+    description: |
+      MENTIONS is the environment variable storing the name for the slack context.
+
+executor: aws-cli/default
 
 steps:
+  - checkout
   - run:
-      name: "Not Yet Implemented"
-      command: echo "Not Yet Implemented"
+      name: "Verify Parameters"
+      command: >
+        if [ -z "$SLACK_MENTIONS" ]; then
```

```suggestion
        if [ -z "$MENTIONS" ]; then
```
This is the envvar in our slack context